x-pack/plugins/lists/public/exceptions/hooks/use_exception_list.ts

```diff
@@ -38,150 +41,153 @@ export const useExceptionList = ({
     perPage: 20,
     total: 0,
   },
-  filterOptions = {
-    filter: '',
-    tags: [],
-  },
+  filterOptions,
+  showDetectionsListsOnly,
+  showEndpointListsOnly,
+  matchFilters,
   onError,
   onSuccess,
 }: UseExceptionListProps): ReturnExceptionListAndItems => {
-  const [exceptionLists, setExceptionLists] = useState<ExceptionList[]>([]);
   const [exceptionItems, setExceptionListItems] = useState<ExceptionListItemSchema[]>([]);
   const [paginationInfo, setPagination] = useState<Pagination>(pagination);
-  const fetchExceptionList = useRef<Func | null>(null);
+  const fetchExceptionListsItems = useRef<Func | null>(null);
   const [loading, setLoading] = useState(true);
-  const tags = useMemo(() => filterOptions.tags.sort().join(), [filterOptions.tags]);
-  const listIds = useMemo(
-    () =>
+  const { ids, namespaces } = useMemo(
+    (): { ids: string[]; namespaces: NamespaceType[] } =>
       lists
-        .map((t) => t.id)
-        .sort()
-        .join(),
-    [lists]
+        .filter((list) => {
+          if (showDetectionsListsOnly) {
+            return list.type === 'detection';
+          } else if (showEndpointListsOnly) {
+            return list.type === 'endpoint';
+          } else {
+            return true;
+          }
+        })
+        .reduce<{ ids: string[]; namespaces: NamespaceType[] }>(
+          (acc, { listId, namespaceType }) => ({
+            ids: [...acc.ids, listId],
+            namespaces: [...acc.namespaces, namespaceType],
+          }),
+          { ids: [], namespaces: [] }
+        ),
+    [lists, showDetectionsListsOnly, showEndpointListsOnly]
+  );
+  const filters = useMemo(
+    (): FilterExceptionsOptions[] =>
+      matchFilters && filterOptions.length > 0 ? ids.map(() => filterOptions[0]) : filterOptions,
+    [matchFilters, filterOptions, ids]
+  );
+  const idsAsString = useMemo((): string => ids.join(), [ids]);
+  const namespacesAsString = useMemo((): string => namespaces.join(), [namespaces]);
+  const filterAsString = useMemo(
+    (): string => filterOptions.map(({ filter }) => filter).join(','),
```

Comment:
nit: I think join's default should be `,` like the one above it. I would use the default if you already started to, so it keeps things consistent.

Nit though, no changes required on nit's or follow up's or ignore ... All fine things.

Reply:
Done!